src/ess/sans/contrib.py

```diff
@@ -0,0 +1,47 @@
+# Generic helpers that may end up as contributions to scipp if cleaned up
+import scipp as sc
+
+
+def midpoints(var, dim):
+    """
+    Utility function for finding middle points of variable
+    """
+    return 0.5 * (var[dim, 1:] + var[dim, :-1])
+
+
+def to_bin_centers(d, dim):
+    """
+    Utility function for setting centers of bins
+    """
+    d.coords[dim] = midpoints(d.coords[dim], dim)
+
+def to_bin_edges(d, dim):
+    """
+    Utility function for setting bin edges
+    """
+    centers = d.coords[dim]
+    del d.coords[dim]
+    first = 1.5 * centers[dim, 0] - 0.5 * centers[dim, 1]
+    last = 1.5 * centers[dim, -1] - 0.5 * centers[dim, -2]
+    bulk = midpoints(centers, dim)
+    edges = sc.concatenate(first, bulk, dim)
+    edges = sc.concatenate(edges, last, dim)
+    d.coords[dim] = edges
```

Comment:
`sc.concatenate` is deprecated, use
```suggestion
    d.coords[dim] = sc.concat([first, bulk, last], dim)
```
unless you are stuck with and old scipp for some reason.